src/test/java/bio/terra/workspace/integration/WorkspaceIntegrationTest.java

```diff
@@ -28,24 +29,91 @@
 @TestPropertySource("classpath:application-integration-test.properties")
 public class WorkspaceIntegrationTest {
 
-  @Autowired private WorkspaceManagerTestClient workspaceManagerTestClient;
+  // TODO: As this class grows, consider if it's worth breaking down these workspace tests into
+  //  different class files based on the the type of workspace action (Create, Get, Delete, etc).
 
+  @Autowired private WorkspaceManagerTestClient workspaceManagerTestClient;
   @Autowired private TestUtils testUtils;
-
   @Autowired private TestConfiguration testConfig;
-
   @Autowired private AuthService authService;
+  private final String TAG_NEEDS_CLEANUP = "needs-cleanup";
+  private static final Logger logger = LoggerFactory.getLogger(WorkspaceIntegrationTest.class);
+
+  // Variable for workspace Id's to clean up in tearDown(TestInfo testInfo) method, after
+  // running applicable test method(s)
+  private UUID workspaceIdToCleanup;
```

Comment:
I'm worried about using a single mutable variable here for use across multiple tests. If we ever switched to using `@TestInstance(Lifecycle.PER_CLASS)`, this could cause consistency problems where multiple tests/teardown operations overwrite each other. I suggest this move to a threadsafe/concurrent collection - probably a `ConcurrentHashMap`, where the key can be a test name (maybe `TestInfo.getTestMethod()`?) and the value can be a list of workspace Ids to clean up. (We should also support a list of workspaces that need cleaning up, not just a single one; tests may need to create/clean up multiple workspaces).

Is this worth it? If this is overkill, at minimum add a comment along the lines of "if we switch TestInstance modes this variable will be a problem".

Reply:
I think it's definitely worth it. This variable is workable but not ideal, and I should have noted that.
I've implemented a `ConcurrentHashMap`.
`TestInfo.getDisplayName()` returns a simpler value (i.e. strips additional metadata that may not be needed).

Reply:
One more thing: I don't think it's necessarily to remove the Id's from the map post-cleanup, but let me know if you disagree.